processor_function/shared/cnab_builder.py

```diff
@@ -113,9 +116,31 @@ def deploy_aci(self):
 
         credential = AzureIdentityCredentialAdapter()
         aci_client = ContainerInstanceManagementClient(credential, self._subscription_id)
+
+        service_bus = ServiceBus()
+        service_bus.send_status_update_message(self._id, "ACI Deployment started", "Deploying ACI container: " + self._container_group_name)
+
         result = aci_client.container_groups.create_or_update(self._resource_group_name, self._container_group_name,
                                                               group)
 
         while result.done() is False:
             logging.info('-- Deploying -- ' + self._container_group_name + " to " + self._resource_group_name)
             time.sleep(1)
+
+        service_bus.send_status_update_message(self._id, "ACI container deployed", "")
+
+        logs = aci_client.container.list_logs(self._resource_group_name, self._container_group_name, self._container_group_name)
+
+        while "Error" not in logs.content and "Success" not in logs.content:
+            time.sleep(5)
+            logs = aci_client.container.list_logs(self._resource_group_name, self._container_group_name, self._container_group_name)
+            if "Error" in logs.content:
+                service_bus.send_status_update_message(self._id, "Deployment failed", logs.content)
```

Comment:
we probably dont want to send all the contents on the service bus

Reply:
Well, it's complicated :) How much should we send? The log is just a dump from ACI command line, which might contain a dump from TF failure. So essentially ~20 lines up from last line might be error information. Should we send just the last line to the cosmos, and the rest to LogAnalytics?

Reply:
We should send the reason for failure. This message is essentially shown to the end user when they query a workspace. We don't want TF and porter errors in their full glory being shown to the API user.

Reply:
Let's spin this into another stand-alone task (Refine error message reporting from Porter); we don't know the exact reason for failure at this point.